lib/datastore/transaction.js

```diff
@@ -85,16 +85,18 @@ nodeutil.inherits(Transaction, DatastoreRequest);
  *     a transaction.
  *
  * @example
- * transaction.begin(function(err) {
- *   // Perform Datastore operations as usual.
- *   transaction.get(dataset.key(['Company', 123]), function(err, entity) {
- *     // Commit the transaction.
- *     transaction.finalize(function(err, apiResponse) {});
+ * dataset.runInTransaction(function(transaction, done) {
+ *   transaction.begin(function(err) {
```

Comment:
I'm getting mixed up if this PR is trying to solve #623. It looks like you wanted to because of wrapping things in `dataset.runInTransaction` and some doc wording changes. If you don't want to solve it in this PR, that's fine, you can leave as is and we can resolve after this merges.